YIF.Core.Service/Concrete/Services/IoEAdminService.cs

```diff
@@ -245,5 +255,50 @@ public async Task<ResponseApiModel<DescriptionResponseApiModel>> DeleteIoEModera
 
             return result.Set(new DescriptionResponseApiModel(_resourceManager.GetString("IoEModeratorIsDeleted")), true);
         }
+
+        public async Task<ResponseApiModel<DescriptionResponseApiModel>> AddLectorToIoE(
+            [NotNull] string userId,
+            [NotNull] EmailApiModel email,
+            [NotNull] HttpRequest request)
+        {
+            var result = new ResponseApiModel<DescriptionResponseApiModel>();
+            var dbUser = new DbUser
+            {
+                Email = email.UserEmail,
+                UserName = email.UserEmail
+            };
+
+            var ioEId = (await _institutionOfEducationAdminRepository.GetByUserId(userId)).InstitutionOfEducationId;
+            var searchUser = await _userManager.FindByEmailAsync(email.UserEmail);
+            if (searchUser == null)
+            {
+                var registerResult = await _userRepository.Create(dbUser, null, null, ProjectRoles.Lecture);
+                if (registerResult != string.Empty)
+                {
+                    throw new BadRequestException($"{_resourceManager.GetString("UserCreationFailed")}: {registerResult}");
+                }
+
+                var resultResetPasswordByEmail = await _userService.ResetPasswordByEmail(email.UserEmail, request);
+                if (!resultResetPasswordByEmail.Success)
+                {
+                    throw new BadRequestException($"{_resourceManager.GetString("ResetPasswordByEmailFailed")}: {resultResetPasswordByEmail.Message}");
+                }
+            }
+
+            else
+            {
+                var lectorExist = await _lectorRepository.GetByUserId(searchUser.Id, ioEId);
+                if (lectorExist != null)
+                {
+                    throw new BadRequestException(_resourceManager.GetString("IoEAlreadyHasLector"));
+                }
+                throw new BadRequestException(_resourceManager.GetString("UserAlreadyExists"));
+            }
+
+            var newLector = new Lecture { InstitutionOfEducationId = ioEId, UserId = dbUser.Id };
+            await _lectorRepository.Add(newLector);
+
+            return result.Set(new DescriptionResponseApiModel(_resourceManager.GetString("LectorWasAdded")), true);
+        }
```

Comment:
this method looks too difficult, maybe you should replace some parts of functionality to the other methods?

Reply:
hmm i have realised such method as in request #138 and it was ok for u, but u don't like realisation in my request.
![image](https://user-images.githubusercontent.com/61685799/119173800-66e02080-ba70-11eb-9445-af3320818266.png)